boards/Makefile.common

```diff
@@ -391,32 +279,22 @@ show-target:
 	$(info $(TARGET))
 
 # This rule is a copy of the rule used to build the release target, but `cargo
-# rustc` has been replaced with `cargo bloat`. `cargo bloat` replicates the
-# interface of `cargo build`, rather than `cargo rustc`, so we need to move
-# `RUSTC_FLAGS_FOR_BIN` into the `RUSTFLAGS` environment variable. This only
-# means that cargo cannot reuse built dependencies built using `cargo bloat`.
-# See the discussion on `RUSTC_FLAGS_FOR_BIN` above for additional details. To
-# pass additional flags to `cargo bloat`, populate the CARGO_BLOAT_FLAGS
-# environment variable, e.g. run `CARGO_BLOAT_FLAGS=--crates make cargobloat`
+# rustc` has been replaced with `cargo bloat`.
 .PHONY: cargobloat
 cargobloat:
-	$(Q)$(CARGO) install cargo-bloat > /dev/null 2>&1 ||  echo 'Error: Failed to install cargo-bloat'
-	$(Q)RUSTFLAGS="$(RUSTC_FLAGS_TOCK) $(RUSTC_FLAGS_FOR_BIN)" CARGO_FLAGS="-Z build-std=core" $(CARGO) bloat $(CARGO_FLAGS_TOCK_NO_BUILD_STD) --bin $(PLATFORM) --release $(CARGO_BLOAT_FLAGS)
+	$(Q)$(CARGO) bloat $(CARGO_FLAGS_TOCK_NO_BUILD_STD) --bin $(PLATFORM) --release $(CARGO_BLOAT_FLAGS)
 
 # To pass additional flags to `cargo cargobloatnoinline`, populate the
 # CARGO_BLOAT_FLAGS environment variable, e.g. run `CARGO_BLOAT_FLAGS=--crates
 # make cargobloatnoinline`
 .PHONY: cargobloatnoinline
 cargobloatnoinline:
-	$(Q)$(CARGO) install cargo-bloat > \dev\null 2>&1 ||  echo 'Error: Failed to install cargo-bloat'
-	$(Q)RUSTFLAGS="$(RUSTC_FLAGS_TOCK) -C inline-threshold=0 $(RUSTC_FLAGS_FOR_BIN)" $(CARGO) bloat $(CARGO_FLAGS_TOCK) --bin $(PLATFORM) --release $(CARGO_BLOAT_FLAGS)
+	$(Q)RUSTFLAGS="-C inline-threshold=0" $(CARGO) bloat $(CARGO_FLAGS_TOCK) --bin $(PLATFORM) --release $(CARGO_BLOAT_FLAGS)
```

Comment:
So this doesn't work any more? How is anyone supposed to use cargo to do something like this?

Reply:
There is actually no point in us having this anymore, since Rust broke the option awhile back and decided to just deprecate it rather than fix it: https://github.com/rust-lang/rust/issues/61088 https://github.com/rust-lang/rust/pull/124712